recipes/irrxml/all/patches/1.2/irrtypes_debug_vs.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/src/irrTypes.h b/src/irrTypes.h
+index aa2254d..8296b8d 100644
+--- a/src/irrTypes.h
++++ b/src/irrTypes.h
+@@ -79,11 +79,7 @@ typedef unsigned short wchar_t;
+ #endif // microsoft compiler
+ 
+ //! define a break macro for debugging only in Win32 mode.
+-#if defined(WIN32) && defined(_MSC_VER) && defined(_DEBUG)
+-#define _IRR_DEBUG_BREAK_IF( _CONDITION_ ) if (_CONDITION_) {_asm int 3}
+-#else 
+ #define _IRR_DEBUG_BREAK_IF( _CONDITION_ )
+-#endif
+ 
+ //! Defines a small statement to work around a microsoft compiler bug.
+ /** The microsft compiler 7.0 - 7.1 has a bug:
+@@ -91,11 +87,7 @@ When you call unmanaged code that returns a bool type value of false from manage
+ the return value may appear as true. See 
+ http://support.microsoft.com/default.aspx?kbid=823071 for details. 
+ Compiler version defines: VC6.0 : 1200, VC7.0 : 1300, VC7.1 : 1310, VC8.0 : 1400*/
+-#if defined(WIN32) && defined(_MSC_VER) && (_MSC_VER > 1299) && (_MSC_VER < 1400)
+-#define _IRR_IMPLEMENT_MANAGED_MARSHALLING_BUGFIX __asm mov eax,100

```

Comment:
not sure about this one, but it might be better to check for **_I_X86** (as inline asm allowed for x86 MSVC)

Reply:
I gonna check it with windows x86